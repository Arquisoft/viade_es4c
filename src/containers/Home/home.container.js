import React from "react";
import {Image} from "react-bootstrap";

import "./home.container.css";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import {Spacer} from "../../components";

/**
 * Home component that returns the homepage
 */
export const HomeComponent = () => {
    return (
        <Container>
            {/* Logo row */}
            <Row>
                <Col xs={12}></Col>
                <Col xs={12} md={8} className="logo-div home-div">
                    <Image src="/viade_es4c/img/logo.svg" alt="Viade"/>
                </Col>
                <Col xs={12}></Col>
            </Row>
            {/* Description row */}
            <Row>
                <Col xs={12}></Col>
                <Col xs={12} md={8} className="home-div">
                    <p>Viade is a Solid project developed by third year students of the University of Oviedo
                        Software Engineering degree in the Software Architecture subject. It offers a route sharing
                        application following the SOLID principles. This is a site where all your data always remains
                        yours. <a href="https://github.com/Arquisoft/viade_es4c">Developed by team es4c</a>.</p>
                </Col>
                <Col xs={12}></Col>
            </Row>
            {/* What you can do row */}
            <Row>
                <Col xs={12}></Col>
                <Col xs={6} md={2} className="home-div">
                    <Image src="/viade_es4c/img/home/uploadroute.svg" alt="Upload a route" className="fill-image"/>
                    <p>Upload a new route</p>
                </Col>
                <Col xs={6} md={2} className="home-div">
                    <Image src="/viade_es4c/img/home/uploadmedia.svg" alt="Upload images and videos" className="fill-image"/>
                    <p>Complete it with images and videos</p>
                </Col>
                <Col xs={6} md={2} className="home-div">
                    <Image src="/viade_es4c/img/home/shareroute.svg" alt="Share it" className="fill-image"/>
                    <p>Share it with whoever you want</p>
                </Col>
                <Col xs={6} md={2} className="home-div">
                    <Image src="/viade_es4c/img/home/keepyours.svg" alt="Keep the ownership of your data" className="fill-image"/>
                    <p>And keep the ownership of all this</p>
                </Col>
                <Col xs={12}></Col>
            </Row>
            {/* Empty rows */}
            <Row>
                <Col><Spacer/></Col>
            </Row>
            <Row>
                <Col><Spacer/></Col>
            </Row>
            {/* Solid/UniOvi/Inrupt row */}
            <Row>
                <Col xs={12}></Col>
                <Col xs={4} md={2} className="home-div logo-div">
                    <Image src="/viade_es4c/img/Solid.svg" alt="Solid" className="fill-image"/>
                </Col>
                <Col xs={4} md={3} className="logo-div home-div">
                    <Image src="/viade_es4c/img/home/uniovi.jpg" alt="Universidad de Oviedo"/>
                </Col>
                <Col xs={4} md={2} className="home-div logo-div">
                    <Image src="/viade_es4c/img/inrupt.svg" alt="Inrupt" className="fill-image"/>
                </Col>
                <Col xs={12}></Col>
            </Row>
        </Container>
    );
};

export default HomeComponent;
