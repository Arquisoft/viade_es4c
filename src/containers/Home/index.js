import {HomeComponent} from "./home.container";

export default HomeComponent;
