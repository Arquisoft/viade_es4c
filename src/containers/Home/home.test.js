import React from 'react';
import { shallow } from 'enzyme';
import Home from "./home.container";

describe('Home', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home debug />);
  
    expect(component).toMatchSnapshot();
  });
});
