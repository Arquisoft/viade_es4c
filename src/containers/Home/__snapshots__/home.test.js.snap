// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly in "debug" mode 1`] = `
<Container
  fluid={false}
>
  <Row
    noGutters={false}
  >
    <Col
      md={2}
      xs={12}
    />
    <Col
      className="logo-div home-div"
      md={8}
      xs={12}
    >
      <Image
        alt="Viade"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/logo.svg"
        thumbnail={false}
      />
    </Col>
    <Col
      md={8}
      xs={12}
    />
  </Row>
  <Row
    noGutters={false}
  >
    <Col
      md={2}
      xs={12}
    />
    <Col
      className="home-div"
      md={8}
      xs={12}
    >
      <p>
        Viade is a Solid project developed by third year students of the University of Oviedo Software Engineering degree in the Software Architecture subject. It offers a route sharing application following the SOLID principles. This is a site where all your data always remains yours.
         
        <a
          href="https://github.com/Arquisoft/viade_es4c"
        >
          Developed by team es4c
        </a>
        .
      </p>
    </Col>
    <Col
      md={2}
      xs={12}
    />
  </Row>
  <Row
    noGutters={false}
  >
    <Col
      md={2}
      xs={12}
    />
    <Col
      className="home-div"
      key="upload"
      md={2}
      xs={6}
    >
      <Image
        alt="Upload a route"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/home/uploadroute.svg"
        thumbnail={false}
      />
      <p>
        Upload a new route
      </p>
    </Col>
    <Col
      className="home-div"
      key="media"
      md={2}
      xs={6}
    >
      <Image
        alt="Upload images and videos"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/home/uploadmedia.svg"
        thumbnail={false}
      />
      <p>
        Complete it with images and videos
      </p>
    </Col>
    <Col
      className="home-div"
      key="share"
      md={2}
      xs={6}
    >
      <Image
        alt="Share it"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/home/shareroute.svg"
        thumbnail={false}
      />
      <p>
        Share it with whoever you want
      </p>
    </Col>
    <Col
      className="home-div"
      key="keep"
      md={2}
      xs={6}
    >
      <Image
        alt="Keep the ownership of your data"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/home/keepyours.svg"
        thumbnail={false}
      />
      <p>
        And keep the ownership of your data
      </p>
    </Col>
    <Col
      md={2}
      xs={12}
    />
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <Spacer />
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <Spacer />
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col
      md={3}
      xs={12}
    />
    <Col
      className="home-div logo-div"
      key="solid"
      md={2}
      xs={4}
    >
      <Image
        alt="Solid"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/Solid.svg"
        thumbnail={false}
      />
      <p />
    </Col>
    <Col
      className="home-div logo-div"
      key="uniovi"
      md={2}
      xs={4}
    >
      <Image
        alt="University of Oviedo"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/home/uniovi.jpg"
        thumbnail={false}
      />
      <p />
    </Col>
    <Col
      className="home-div logo-div"
      key="inrupt"
      md={2}
      xs={4}
    >
      <Image
        alt="inrupt"
        className="fill-image"
        fluid={false}
        rounded={false}
        roundedCircle={false}
        src="/img/inrupt.svg"
        thumbnail={false}
      />
      <p />
    </Col>
    <Col
      md={3}
      xs={12}
    />
  </Row>
</Container>
`;
